tests/app/notifications/test_process_notification.py

```diff
@@ -135,24 +135,28 @@ def test_persist_notification_with_optionals(sample_job, sample_api_key, mocker)
     assert persisted_notification.reference is None
 
 
-@pytest.mark.parametrize('research_mode, queue, notification_type, key_type',
-                         [(True, 'research-mode', 'sms', 'normal'),
-                          (True, 'research-mode', 'email', 'normal'),
-                          (True, 'research-mode', 'email', 'team'),
-                          (False, 'send-sms', 'sms', 'normal'),
-                          (False, 'send-email', 'email', 'normal'),
-                          (False, 'send-sms', 'sms', 'team'),
-                          (False, 'research-mode', 'sms', 'test')])
+@pytest.mark.parametrize('research_mode, requested_queue, expected_queue, notification_type, key_type',
+                         [(True, None, 'research-mode', 'sms', 'normal'),
+                          (True, None, 'research-mode', 'email', 'normal'),
+                          (True, None, 'research-mode', 'email', 'team'),
+                          (False, None, 'send-sms', 'sms', 'normal'),
+                          (False, None, 'send-email', 'email', 'normal'),
+                          (False, None, 'send-sms', 'sms', 'team'),
+                          (False, None, 'research-mode', 'sms', 'test'),
+                          (True, 'notify', 'research-mode', 'email', 'normal'),
+                          (False, 'notify', 'notify', 'sms', 'normal'),
+                          (False, 'notify', 'notify', 'email', 'normal'),
+                          (False, 'notify', 'research-mode', 'sms', 'test')])
 def test_send_notification_to_queue(notify_db, notify_db_session,
```

Comment:
this test could probably be done without DB (probably just a `namedtuple('Notification', ['id', 'notification_type', 'key_type'])`), which would be a nice test perf win considering all the different parametrized test cases here